x86_mm256_sad_epu8 intrinsic has incorrect return type

According to [intel documentation](https://software.intel.com/sites/landingpage/IntrinsicsGuide/#techs=AVX,AVX2&expand=2800,362,4676,4678,2880,532,533,4726,4535&text=sad) the return type should be `64x4` but rustc expects a `8x32`

Test case:

```rust
#![feature(platform_intrinsics)]
#![feature(repr_simd)] 

#[repr(simd)]
struct u8x32(u8,u8,u8,u8,u8,u8,u8,u8,u8,u8,u8,u8,u8,u8,u8,u8,u8,u8,u8,u8,u8,u8,u8,u8,u8,u8,u8,u8,u8,u8,u8,u8,);
#[repr(simd)]
struct u64x4(u64,u64,u64,u64,);

extern "platform-intrinsic" {
 fn x86_mm256_sad_epu8(x: u8x32, y: u8x32) -> u64x4;
}

fn main() {}
```

Compiler error:

```
error[E0442]: intrinsic return value has wrong type: found vector with length 4, expected length 32
  --> src/main.rs:10:2
   |
10 |  fn x86_mm256_sad_epu8(x: u8x32, y: u8x32) -> u64x4;
   |  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

--- a/src/etc/platform-intrinsics/x86/avx2.json
+++ b/src/etc/platform-intrinsics/x86/avx2.json
@@ -174,7 +174,7 @@
             "intrinsic": "256_sad_epu8",
             "width": [256],
             "llvm": "psad.bw",
-            "ret": "u8",
+            "ret": "u64",
             "args": ["0", "0"]
         },
         {

--- a/src/librustc_platform_intrinsics/x86.rs
+++ b/src/librustc_platform_intrinsics/x86.rs
@@ -354,7 +354,7 @@ pub fn find(name: &str) -> Option<Intrinsic> {
         },
         "_mm256_sad_epu8" => Intrinsic {
             inputs: { static INPUTS: [&'static Type; 2] = [&::U8x32, &::U8x32]; &INPUTS },
-            output: &::U8x32,
+            output: &::U64x4,
             definition: Named("llvm.x86.avx2.psad.bw")
         },
         "_mm256_shuffle_epi8" => Intrinsic {

